app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,793 @@
+            return <tr key={i}>{this.insideEachRow(dataObj)}</tr>;
+          }
+        }.bind(this))
+      );
+    },
+
+    insideEachRow: function (dataObj) {
+      return _.map(dataObj, function (dataVal, dataKey) {
+        return <td key={dataKey} title={dataVal}>{dataVal}</td>;
+      });
+    },
+
+    header: function () {
+      var header = null;
+
+      if (this.props.getHeaderConfig) {
+        header = this.props.meta.fields;
+        return (
+          header.map(function (field, i) {
+            return <th key={i} title={field}>{field}</th>;
+          })
+        );
+      } else {
+        header = this.props.data;
+        return (
+          header.map(function (field, i) {
+           return <th key={i} title={i}>{i}</th>;
+          })
+        );
+      }
+    },
+
+    render: function () {
+      var data = this.eachRow(),
+          header = this.header();
+
+      if (this.props.getPreviewView === 'table') {
+        return (
+          <table className="data-import-table">
+            <tbody>
+              <tr>{header}</tr>
+              {data}
+            </tbody>
+          </table>
+        );
+      } else {
+        return null;
+      }
+    }
+  });
+
+  var JSONView = React.createClass({
+    objectify: function (array) {
+      return _.reduce(array, function (obj, val, i) {
+        obj[i] = val;
+        return obj;
+      }, {});
+    },
+
+    rows: function () {
+      var data = this.props.data;
+
+      if (this.props.isBigFile) {
+        data = this.props.getSmallPreviewOfData;
+      }
+
+      return (
+        data.map(function (dataObj, i) {
+          var obj = this.props.getHeaderConfig ? dataObj :
+            this.objectify(dataObj);
+          if (i < 500) {
+            return (
+              <Components.SimpleDoc
+                id={"<UUID>_" + i}
+                content={JSON.stringify(obj, null, ' ')}
+                key={i} />
+            );
+          }
+        }.bind(this))
+      );
+    },
+
+    render: function () {
+      if (this.props.getPreviewView === "json") {
+        return (
+          <div id="doc-list" className="json-view">
+            {this.rows()}
+          </div>
+        );
+      } else {
+        return null;
+      }
+    }
+  });
+
+  var Footer = React.createClass({
+    getInitialState: function () {
+      return {
+        targetDB: this.props.getAllDBs[0],
+        selectExistingDB: true
+      };
+    },
+
+    startOverButton: function () {
+      return (
+        <a className="start-import-over-link footer-button"
+          onClick={this.startover}>
+          <span className="fonticon icon-repeat"></span>
+            Start Over
+        </a>
+      );
+    },
+
+    startover: function () {
+      Actions.dataImporterInit(true);
+    },
+
+    getExistingDBList: function () {
+      var allDBs = this.props.getAllDBs,
+          options = [],
+          setTargetDB = this.setTargetDB;
+
+      _.each(allDBs, function (dbName, i) {
```

Comment:
You could drop the `options` var here and just return _.map() where each iteration returns the object with name + onClick.